openshift_tools/monitoring/hawk_sender.py

```diff
@@ -0,0 +1,190 @@
+            print unique_metric
+        print "==============================\n"
+
+    def parse_config(self):
+        """ parse default config file """
+
+        if not self.config:
+            if not os.path.exists(self.config_file):
+                raise HawkSenderException(self.config_file + " does not exist.")
+            self.config = yaml.load(file(self.config_file))
+
+    def get_default_host(self):
+        """ get the 'host' value from the config file """
+        self.parse_config()
+
+        return self.config['host']['name']
+
+    def get_default_hawk_connecton(self):
+        """ get the values and create a hawk_connection """
+
+        self.parse_config()
+        hawk_server = self.config['hawk']['url']
+        hawk_user = self.config['hawk']['user']
+        hawk_password = self.config['hawk']['pass']
+        hawk_ssl_verify = self.config['hawk'].get('ssl_verify', False)
+        hawk_debug = self.config['hawk'].get('debug', False)
+        hawk_active = self.config['hawk'].get('active', True)
+
+        if isinstance(hawk_ssl_verify, str):
+            hawk_ssl_verify = (hawk_ssl_verify == 'True')
+
+        if self.debug:
+            hawk_debug = self.debug
+        elif isinstance(hawk_debug, str):
+            hawk_debug = (hawk_debug == 'True')
+
+        hawk_connection = HawkConnection(url=hawk_server,
+                                         user=hawk_user,
+                                         password=hawk_password,
+                                         ssl_verify=hawk_ssl_verify,
+                                         debug=hawk_debug,
+                                         active=hawk_active,
+                                        )
+
+        return hawk_connection
+
+    def add_heartbeat(self, heartbeat, host=None):
+        """ create a heartbeat unique metric to send to hawk """
+
+        if not host:
+            host = self.host
+
+        hb_metric = UniqueMetric.create_heartbeat(host,
+                                                  heartbeat.templates,
+                                                  heartbeat.hostgroups,
+                                                 )
+        self.unique_metrics.append(hb_metric)
+
+    def add_zabbix_keys(self, zabbix_keys, host=None, synthetic=False):
+        """ create unique metric from zabbix key value pair """
+
+        if synthetic and not host:
+            host = self.config['synthetic_clusterwide']['host']['name']
+        elif not host:
+            host = self.host
+
+        zabbix_metrics = []
+
+        for key, value in zabbix_keys.iteritems():
+            zabbix_metric = UniqueMetric(host, key, value)
+            zabbix_metrics.append(zabbix_metric)
+
+        self.unique_metrics += zabbix_metrics
+
+    # Allow for 6 arguments (including 'self')
+    # pylint: disable=too-many-arguments
+    def add_zabbix_dynamic_item(self, discovery_key, macro_string, macro_array, host=None, synthetic=False):
```

Comment:
Dynamic items in zabbix are a way to send arbitrary data into zabbix in a flexible way.

For Hawk, all metrics are dynamic and flexible. So this shouldn't be needed at all. I'd remove it. I understand this has implications because we're trying to have a similar interface between the two, so maybe further discussion is necessary here.

This might just need to turn into something that creates normal hawkular metrics, but with extra tags. Yeah, more discussion is needed around this one.

At the very least, the name should be changed to not include `zabbix`. Suggested name `add_dynamic_metric`.